src/main/java/com/netchill/webservices/api/GenreWs.java

```diff
@@ -29,9 +32,15 @@ public GenreWs(ConfigurationService configurationService, GenreService genreServ
     }
 
     @GET
-    @Path("/id")
-    public Genre getGenreById(@QueryParam("id") Long id) {
-        return this.genreService.getGenreById(id);
+    @Path("/{id}")
+    public Optional<Genre> getGenreById(@PathParam("id") Long id) {
+        Optional<Genre> result = this.genreService.getGenreById(id);
+
+        if (result == null) {
+            throw new WsException(NetchillWsError.RESOURCE_NOT_FOUND);
+        }
```

Comment:
result ne sera jamais null, c'est un optional.

il faut faire :

```suggestion
        Optional<Genre> result = this.genreService.getGenreById(id);

        if (result.isEmpty()) {
            throw new WsException(NetchillWsError.RESOURCE_NOT_FOUND);
        }
```

Ou 

``` suggestion
        // Ici on récupère le contenu de l'`Optional`, et s'il est vide on met `null` comme valeur par defaut.
        // Dans d'autre situation tu pourrais mettre autre chose comme valeur par defaut.
        Genre result = this.genreService.getGenreById(id).orElse(null);

        if (result == null) {
            throw new WsException(NetchillWsError.RESOURCE_NOT_FOUND);
        }
```


Si tu veux aller plus loin, c'est faisable en une seul instruction :

``` suggestion
        // Ici on récupère le contenu de l'`Optional`, et s'il est vide on  throw une exception.
        return this.genreService.getGenreById(id)
            .orElseThrow(()-> new WsException(NetchillWsError.RESOURCE_NOT_FOUND));
```

Reply:
Quand je fais : 
`return this.genreService.getGenreById(id).orElseThrow(()-> new WsException(NetchillWsError.RESOURCE_NOT_FOUND))`

Ça ne marche pas il faut que je fasse : 

`return Optional.ofNullable(this.genreService.getGenreById(id).orElseThrow(()-> new WsException(NetchillWsError.RESOURCE_NOT_FOUND)));`

Reply:
Pour faire ça je dois pas return un Optional, sinon j'indique Optional.ofNullable devant "this.genreService.getGenreById(id)" ?

Reply:
non jamais !
Les optionals ne sont à utiliser quand dans des returns, pas pour des calcules.

Donc oui il faut modifier `getGenreById`.